src/nanoarrow/common/inline_types.h

```diff
@@ -623,7 +623,8 @@ enum ArrowBufferType {
   NANOARROW_BUFFER_TYPE_UNION_OFFSET,
   NANOARROW_BUFFER_TYPE_DATA_OFFSET,
   NANOARROW_BUFFER_TYPE_DATA,
-  NANOARROW_BUFFER_TYPE_DATA_VIEW
+  NANOARROW_BUFFER_TYPE_VARIADIC_DATA,
```

Comment:
If we are going to introduce new members like this should we think about renaming the existing member to `NANOARROW_BUFFER_TYPE_FIXED_DATA`? I think the distinction can get easily lost